src/main/java/org/example/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+        order.setStatus(Status.PENDING);
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = cartItems.stream().map(cartItem -> {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice().multiply(
+                    BigDecimal.valueOf(cartItem.getQuantity())));
+
+            return orderItem;
+        }).collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+        order.setTotal(calculateTotal(orderItems));
+
+        Order savedOrder = orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        cartItemRepository.deleteAllByShoppingCartId(shoppingCart.getId());
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<OrderResponseDto> getAllUserOrders(Long userId,Pageable pageable) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException(
+                "can`t find user " + userId));
+        List<Order> ordersByUserId = orderRepository.findAllByUserId(user.getId());
+        return ordersByUserId.stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<OrderItemResponseDto> getAllItemsByOrderId(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("can`t find "
+                        + "order by id: " + orderId));
+        Set<OrderItem> orderItems = order.getOrderItems();
+
+        return orderItems.stream()
+                .map(orderItemsMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public OrderItemResponseDto getItemByOrderIdAndItemId(Long orderId, Long itemId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("can`t find "
+                        + "order by id: " + orderId));
+        Set<OrderItem> orderItems = order.getOrderItems();
+        return orderItems.stream().filter(e -> Objects.equals(e.getId(), itemId))
+                .map(orderItemsMapper::toDto).findFirst().get();
+    }
+
+    @Override
+    @Transactional
+    public OrderUpdatedDto changeStatusOfOrderById(Long orderId,
+                                                   OrderStatusUpdateRequestDto request) {
+        Order order = orderRepository.findById(orderId).orElseThrow(() ->
+                new EntityNotFoundException("Order with id: " + orderId + " does not exist"));
+
+        String statusString = request.getStatus().trim().toUpperCase();
+        System.out.println("Received status from request: " + statusString);
```

Comment:
If you want to add logs please use logger

Reply:
+